### PR TITLE
[MSHARED-1090] Update module

### PR DESCRIPTION
**Pull request by cstamas**
Update module:
* parent to POM 36
* Maven level to 3.2.5
* Java level to 8
* drop m-shared-utils (deprecated method call replaced with what deprecation suggested)
* drop legacy APIs like plexus logger, all the logger wrapper, settle on slf4j

---

https://issues.apache.org/jira/browse/MSHARED-1090

### CODE REVIEW

**Comment by jorsol** [comment]
This change should probably bump the minor version.

**Comment by cstamas** [comment]
Of course, but did not want to mix the two (there are a LOT of other changes already requiring it)

**Comment by cstamas** [comment]
Hm, maybe we need NoopMessageSink to retain existing behaviour with defautl ctor... as now it will use sl4j logger and this is behaviour change from before when def ctor used.

**Comment by michael-o** [comment]
Where is this module still used?

**Comment by cstamas** [comment]
maven-jar-plugin for sure, unsure for others, but gh search reveals a LOT of uses across many (non our) plugins

**Comment by cstamas** [comment]
> Why do you I have the feeling that I have seen this with Plexus Utils too?!

This is "already seen" to me as well, no idea where but really looks familiar...

**Comment by slawekjaranowski** [comment]
As @slachiewicz said - why not use slf4j?

There is changed from plexus Logger to slf4j Logger so change will be incompatible.

File: pom.xml [hunk 1]
```diff
@@ -71,24 +72,36 @@
...
     <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
-      <version>4.13.1</version>
+      <version>4.13.2</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-nop</artifactId>
```
Comment (michael-o) [inline pom.xml:103]: Really nop or simple?

**Review by michael-o — CHANGES_REQUESTED** [review]
Please, can we throw out this ugly wrapper around wrappe around SLF4J even if we need to bump the major version?!

**Comment by cstamas** [comment]
Tidied up completely

File: src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java [hunk 2]
```diff
@@ -61,77 +63,31 @@
...
     }
 
     /**
      * Create a new manager instance with the supplied log instance. Verbose flag is set to false.
      *
-     * @param log The mojo log instance
+     * @param logger The log instance
      */
-    public FileSetManager( Logger log )
+    public FileSetManager( Logger logger )
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java:80]: I think this ctor should be deprecated.

File: src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java [hunk 3]
```diff
@@ -285,18 +241,18 @@ public void delete( FileSet fileSet, boolean throwsError )
                 {
                     if ( fileSet.isFollowSymlinks() || !isSymlink( file ) )
                     {
-                        if ( verbose && messages != null )
+                        if ( verbose )
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java:244]: This completely circumvents log levels

File: src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java [hunk 4]
```diff
@@ -285,18 +241,18 @@ public void delete( FileSet fileSet, boolean throwsError )
...
-                            messages.addInfoMessage( "Deleting directory: " + file ).flush();
+                            logger.info( "Deleting directory: " + file );
                         }
 
                         removeDir( file, fileSet.isFollowSymlinks(), throwsError, warnMessages );
                     }
                     else
                     { // delete a symlink to a directory without follow
-                        if ( verbose && messages != null )
+                        if ( verbose )
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java:253]: ditto

File: src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java [hunk 5]
```diff
@@ -316,9 +272,9 @@ public void delete( FileSet fileSet, boolean throwsError )
                 }
                 else
                 {
-                    if ( verbose && messages != null )
+                    if ( verbose )
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java:275]: ditto

File: src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java [hunk 6]
```diff
@@ -380,9 +336,9 @@ private Set<String> findDeletablePaths( FileSet fileSet )
 
     private Set<String> findDeletableDirectories( FileSet fileSet )
     {
-        if ( verbose && messages != null )
+        if ( verbose )
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java:339]: ditto

File: src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java [hunk 7]
```diff
@@ -398,31 +354,31 @@ private Set<String> findDeletableDirectories( FileSet fileSet )
 
         if ( !fileSet.isFollowSymlinks() )
         {
-            if ( verbose && messages != null )
+            if ( verbose )
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java:357]: ditto

File: src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java [hunk 8]
```diff
@@ -437,9 +393,9 @@ private Set<String> findDeletableDirectories( FileSet fileSet )
 
     private Set<String> findDeletableFiles( FileSet fileSet, Set<String> deletableDirectories )
     {
-        if ( verbose && messages != null )
+        if ( verbose )
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java:396]: ditto

File: src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java [hunk 9]
```diff
@@ -455,31 +411,31 @@ private Set<String> findDeletableFiles( FileSet fileSet, Set<String> deletableDi
 
         if ( !fileSet.isFollowSymlinks() )
         {
-            if ( verbose && messages != null )
+            if ( verbose )
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java:414]: ditto

**Comment by cstamas** [comment]
re verbose: disagree. This is "verbose", will tell more, that's all. Log level is INFO

**Comment by cstamas** [comment]
verbose is orthogonal to log level. That's all IMHO

File: src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java (hunk 2 above)
Comment (cstamas) [reply src/main/java/org/apache/maven/shared/model/fileset/util/FileSetManager.java:80]: Disagree, this ctor controls what logger this class will use.

**Comment by cstamas** [comment]
General: this is a utility class, is NOT a component.

**Review by michael-o — APPROVED** [review]

**Pull request merged**